packages/material/src/layouts/MaterialCategorizationLayout.tsx

```diff
@@ -92,7 +92,8 @@ export class MaterialCategorizationLayoutRenderer extends RendererComponent<
       uischema,
       visible,
       enabled,
-      selected
+      selected,
+      ajv
```

Comment:
Please hand in this ajv with an extra HOC

Reply:
Should this extra HOC come from /core/src/util/renderer.ts?

How should it be named? I'm trying to figure out the pattern to use but struggling.

Reply:
It's fine to just place it in the `MaterialCategorizationLayoutRenderer` itself for now. You can take a look at `withVanillaControlProps` as a guide. It should look something like this (untested):
```
export const withAjv = (Component) => (props) => {
  const ctx = useJsonForms();
  return <Component {...props} {ajv: ctx.core.ajv} />;
}
```

Reply:
Oh! I can get it with `withJsonFormsContext`, then just do a little `getAjv({ jsonforms: { ...ctx } })`

Reply:
I didn't see this comment until now. The HOC is needed in 2 files in the material package, so I put it in the `util/layout.tsx` file. If you'd like it somewhere else I can move it.

Also, I set up it's types so it's able to correctly return the prop types for the outputted component. This pattern might be helpful in the `withJsonFormsContext` HOC as well if you want to try it out there.

Reply:
Oh, and just a heads up: `withAjv(withJsonFormsLayoutProps(MaterialCategorizationLayoutRenderer))` doesn't work because `withJsonFormsLayoutProps` returns a component with type: `React.ComponentType<OwnPropsOfLayout>`. It would have to be a generic which returns a component of type `React.ComponentType<OwnPropsOfLayout & T>` where T are any props that are additional to the OwnPropsOfLayout + any props that are injected and abstracted away.

This is a simple issue to work around, however. It just means you have to do it in this order:

`withJsonFormsLayoutProps(withAjvProps(MaterialCategorizationStepperLayoutRenderer));`